### DISCUSSION TITLE
Refactor Library

### DISCUSSION THREAD

**Discussion by Andre601** [Chew/JDA-Chewtils#41] [Ideas]
This was discussed/mentioned before but the entire lib (or at least the entirety of the Pagination stuff) should be refactored to be improved.

When this is done, would I suggest to put the new code under a different path, to separate it from the old section, so that people know they have to switch over. Maybe have a brief support time for the old stuff with Deprecation?

**Comment by Chew** [comment]
Definitely agree, though if we want to completely refactor something it may be best to move away from the `com.jagrosh` package to something else (probably `pw.chew`)?

Obviously will want deprecations on the old paginators, probably a `@Deprecated`, and offer they switch to the new paginator. No point in removing the old paginators until more people adopt off of JDA-U for backward compatibility or whatever.

As for the other bits, we may want to do the same with Command and CommandEvent, at least with CommandEvent. That file's messy as hell and uses the old reply logic.

Dunno if we want a `this.useNewEvent` or something that calls a new CommandEvent class where the old Is deprecated? The Command class seems fine atm as is, though.

**Comment by Andre601** [comment]
> Definitely agree, though if we want to completely refactor something it may be best to move away from the `com.jagrosh` package to something else (probably `pw.chew`)?

I did mention this:
> When this is done, would I suggest to put the new code under a different path, to separate it from the old section,

**Comment by freya022** [comment]
![yes](https://user-images.githubusercontent.com/41875020/127414253-a2b4bfdd-9a1f-4dee-a64a-f70911c2b19d.jpg)
